### PR TITLE
FormResolveConflict: Allow open without cmd/path

### LINKED ISSUE #8666: Git Ext incorrectly diagnoses bad diff/merge tool configuration, will not provide conflict resolution window
## Current behaviour

When there is a merge conflict, Git Extensions does a check to see if your merge tool is properly configured before even allowing you to go to the conflict resolution window. Specifically the issue is that Git Extensions apparently is requiring the `.gitconfig` to have a "path =" property within the configured diff and/or merge tool section, when Git itself does not require this, as long as the "cmd =" property can be run from the command line (usually, because the executable being run is already on the path).

The following window pops up upon attempting to enter the conflict resolution window from any entry vector:
image

This is made even more confusing by the fact that the main settings page says that the diff and merge tools are configured correctly:
image

### Issue 1:

So, Git Extensions is requiring a stricter configuration than Git itself does, apparently only as a restriction to the conflict resolution window, as though a different Git configuration check for the merge tool is being made than what is done on the main settings page.

This additional requirement of having the "path" property present makes Git Extensions incompatible with other existing tools and procedures that setup the `.gitconfig` file according to Git's requirements, not this addition constraint from Git Extensions.

Furthermore, this is extremely confusing for new users (and, actually, old users), given the conflict between the main settings page correctly indicating that tools are configured without error, and the conflict resolution window check using a different standard.

### Issue 2:

Secondarily, this check for mergetool configuration is made before even allowing the user into the conflict resolution window, despite the fact that there are actions the user can take in that window that do not launch/run the merge tool, which the user can take even if the mergetool is incorrectly configured.

For example, in that window, one can choose to abort the merge, and can manually resolve conflicts by choosing the entire incoming or existing file as-is (on a per-file basis), without launching a manual or visual merge tool.

In addition to fixing issue 1, the check for configuration should only be made before needing to actually launch a diff/merge tool, so that other viable options which don't require a merge tool can still be carried out, so that unrelated functionality is not unnecessarily restricted.

### Issue 1 - incorrectly requiring a "path" property for merge/diff tools

Expect Git Extensions to not overly constrain configuration checks to requiring a "path =" property in `.gitconfig` for merge tools, and especially not to contradict it's determination in the main settings page that things are configured correctly.

If the concern is a crash of Git Extensions if it is misconfigured (which I realize was initially happening with v3.x), I expect Git Extensions to either check the "cmd" property for executable separately through parsing when a "path" property is not available, or just try to launch the tool and properly catch and handle the exception if it is generated by Git after attempting to launch.

### Issue 2 - not allowing the user to open the conflict resolution window if merge tool is misconfigured

Move the check for merge tool configuration to the "Launch Merge Tool" button or into whatever process actually launches the tool, rather than constraining other unrelated activites.

## Steps to reproduce

1. Change `.gitconfig` current merge (and/or diff?) tool config section so that the "path" property doesn't exist, with a tool that has the location available on the system or user path (i.e. it will launch correctly from Git, regardless of the missing path). Note that you can add the path itself to the "cmd" property, which essentially makes the path property redundant and unnecessary.

1b. After removing the "path" property, Git Extensions will not show any error on the main settings page if the exectuable in the "cmd" property is launchable.

2. Begin a merge that has a conflict.

3. User will be unable to enter the conflict resolution window by any entry vector, being given a window that this cannot be done since the merge tool is misconfigured.

## Did this work in previous version of GitExtensions

Yes - original v3.x versions simply entered the Conflict Resolution window, but that window had a lot of buttons that were generating crashes, including the merge tool button. It appears this check to even enter the conflict resolution window was added to help fix those issues.

That happened a while ago, I just keep running across it with my team occasionally. We've updated our Git instructions to require the path property on config, but this is still not right and I suspect causes significant confusion for other users in this situation.

## Environment

- Git Extensions 3.4.3.9999
- Build d4b0f48bbf3e39a71f5d7ff5231be5678d4aa0f1
- Git 2.29.2.windows.2
- Microsoft Windows NT 10.0.18363.0
- .NET Framework 4.8.4250.0
- DPI 120dpi (125% scaling)

### PR DESCRIPTION
**Pull request by gerhardol**
Fixes #8666

## Proposed changes

Allow opening of FormResolveConflicts also if the mergetool cmd and path is not set.
The explicit button and menu item for GE "enhanced" mergetool handling is disabled if cmd or path is not set.
The popup warning about the configuration was previously shown if path did not exist. It is now shown when the path is set but does not exist (but the form is still shown).

It is allowed to start the form without cmd being set (but path must be set and exist in 3.4).
If cmd or path does not exist, GE falls back to using Git mergetool since 3.4, see 2436c6a64dcad34e06ba9ee62f78f5f5f052b710

Note that git-mergetool has fallback handling if the mergetool, does not exist, see the screenshot.
In next version it is possible to explicitly use a specific mergetool too.

Explicitly disabling the buttons is not sufficient to disable attempts to run "GE mergetool", for instance hotkeys and doubleclicking tries to involve GE mergetool (why the fallback ti git-mergetool is required).
 
### Before

image

### After

image

image

## Test methodology 

manual

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Review by RussKie — COMMENTED** [review]
I can't run the change atm, looking the code I can't seem to understand how we're populating the dropdown. Could you please explain?

File: GitUI/CommandsDialogs/FormResolveConflicts.cs [hunk 1]
```diff
@@ -727,6 +735,11 @@ private string GetShortHash(ConflictedFileData item)
         }
 
         private void ConflictedFiles_SelectionChanged(object sender, EventArgs e)
+        {
+            ConflictedFiles_SelectionChanged();
+        }
+
+        private void ConflictedFiles_SelectionChanged()
```
[the remark this answers is not in the document]
Comment (gerhardol) [reply GitUI/CommandsDialogs/FormResolveConflicts.cs:742]: `UpdateConflictedFilesMenu`

`ConflictedFilesContextMenu` is populated at init, availability of commands changes when selection changes.
It also affects `openMergeToolBtn`.

This PR changes `openMergeToolBtn` and the menu alternative, enabling only when the configured mergetool has both path/cmd that Git do not require.

**Review by mstv — APPROVED** [review]
LGTM but have not run it.

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
Please make up the history and merge at will

**Comment by gerhardol** [comment]
> Please make up the history and merge at will

One minor simplification added, buttons do not have to be disabled from the InitMergetool() result, this is done in UpdateConflictedFilesMenu() anyway. (A change in the fixup that was not fully simplifying the code.)

**Comment by RussKie** [comment]
Looks like we've regressed the UX with this change, now whenever there's a conflict I get a cmd window opened:
image

Can you please have a look?

**Pull request merged**